tests/test_other.py

```diff
@@ -9618,66 +9627,121 @@ def test_minimal_runtime_code_size(self):
+            gzf.write(open(f, 'rb').read())
+          size = os.path.getsize(f_gz)
+          try_delete(f_gz)
+          return size
+
+        obtained_results = {}
+        total_output_size = 0
+        total_expected_size = 0
+        total_output_size_gz = 0
+        total_expected_size_gz = 0
+        for f in outputs:
+          f_gz = f + '.gz'
+          expected_size = expected_results[f] if f in expected_results else float('inf')
+          expected_size_gz = expected_results[f_gz] if f_gz in expected_results else float('inf')
+          size = os.path.getsize(f)
+          size_gz = get_file_gzipped_size(f)
+
+          obtained_results[f] = size
+          obtained_results[f_gz] = size_gz
+          print('size of ' + f + ' == ' + str(size) + ', expected ' + str(expected_size) + ', delta=' + str(size - expected_size) + print_percent(size, expected_size))
+          print('size of ' + f_gz + ' == ' + str(size_gz) + ', expected ' + str(expected_size_gz) + ', delta=' + str(size_gz - expected_size_gz) + print_percent(size_gz, expected_size_gz))
+
+          # Hack: Generated .mem initializer files have different sizes on different
+          # platforms (Windows gives x, CircleCI Linux gives x-17 bytes, my home
+          # Linux gives x+2 bytes..). Likewise asm.js files seem to be affected by
+          # the LLVM IR text names, which lead to asm.js names, which leads to
+          # difference code size, which leads to different relooper choices,
+          # as a result leading to slightly different total code sizes.
+          # TODO: identify what is causing this. meanwhile allow some amount of slop
+          mem_slop = 20
+          if size <= expected_size + mem_slop and size >= expected_size - mem_slop:
+            size = expected_size
+
+          # N.B. even though the test code above prints out gzip compressed sizes, regression testing is done against uncompressed sizes
+          # this is because optimizing for compressed sizes can be unpredictable and sometimes counterproductive
+          total_output_size += size
+          total_expected_size += expected_size
+
+          total_output_size_gz += size_gz
+          total_expected_size_gz += expected_size_gz
+
+        obtained_results['total'] = total_output_size
+        obtained_results['total_gz'] = total_output_size_gz
+        print('Total output size=' + str(total_output_size) + ' bytes, expected total size=' + str(total_expected_size) + ', delta=' + str(total_output_size - total_expected_size) + print_percent(total_output_size, total_expected_size))
+        print('Total output size gzipped=' + str(total_output_size_gz) + ' bytes, expected total size gzipped=' + str(total_expected_size_gz) + ', delta=' + str(total_output_size_gz - total_expected_size_gz) + print_percent(total_output_size_gz, total_expected_size_gz))
+
+        if os.environ.get('EMTEST_REBASELINE'):
+          open(results_file, 'w').write(json.dumps(obtained_results))
```

Comment:
Can you call call `self.assertFileContents(results_file, obtained_results)` rather then duplicating the logic?

Reply:
If you format the JSON nicely the fails will be reported pretty clearly via the output of the diff I think.. so you can remove the check below?

Reply:
That would cause the code to verify both gzip compressed and non-compressed sizes, but I do not want to test gzip compressed sizes, just print them, since strictly optimizing for gzip compressed sizes is not productive. (also there is no guarantee that different systems will produce identical gzip compressed output)

Reply:
I still think why you can replace lines 9739 through 9747 with

```
expected = json.dumps(obtained_results, indent=2) + '\n'
self.assertFileContents(results_file, expected)
```

If you want to print something and not compare it then isn't the best way simply not to include it in the `expected` string.   (Otherwise you will get cases where the tests pass by EMTEST_REBASELINE still produces a diff.. which is a little confusing I think).

The side effect of doing it this way is that the tests will "fail fast", but I actually prefer this over the method of storing the failure on the side and delaying reporting until later.  Plus `EMTEST_REBASELINE` makes it easy to run through all them effectively skipping over the failures.

Reply:
> Otherwise you will get cases where the tests pass by EMTEST_REBASELINE still produces a diff.. which is a little confusing I think).

The .gz sizes are included so that the effect on their size changes can be easily tracked when doing changes locally. We have not tracked them before, and it would be informative to know how they change, we'd lose that if so.

Also using assertFileContents would mean dropping the current allowance for slop that we have now. Neither compiler produces identical results across OSes and systems, for unknown reasons. That would probably require first fixing up all sources of nondeterminism in the compiler backends first.